packages/flutter/lib/src/material/drawer.dart

```diff
@@ -260,17 +260,26 @@ class Drawer extends StatelessWidget {
       namesRoute: true,
       explicitChildNodes: true,
       label: label,
-      child: ConstrainedBox(
-        constraints: BoxConstraints.expand(width: width ?? drawerTheme.width ?? _kWidth),
-        child: Material(
-          color: backgroundColor ?? drawerTheme.backgroundColor ?? defaults.backgroundColor,
-          elevation: elevation ?? drawerTheme.elevation ?? defaults.elevation!,
-          shadowColor: shadowColor ?? drawerTheme.shadowColor ?? defaults.shadowColor,
-          surfaceTintColor: surfaceTintColor ?? drawerTheme.surfaceTintColor ?? defaults.surfaceTintColor,
-          shape: shape ?? (isDrawerStart
-            ? (drawerTheme.shape ?? defaults.shape)
-            : (drawerTheme.endShape ?? defaults.endShape)),
-          child: child,
+      // This prevents notifications from being bubbled up
+      // to the Scaffold.
+      child: NotificationListener<Notification>(
+        onNotification: (Notification notification) {
+          return true;
+        },
+        child: ScrollNotificationObserver(
```

Comment:
I think I know why just adding this did not work. ScrollNotificationObserver is where the notification is actually handled in the context of it bubbling further up the tree. It just returns false after passing the notification to the listeners:

https://github.com/flutter/flutter/blob/f6bc147c91805e7d5e17ce753b1958e19648a9bc/packages/flutter/lib/src/widgets/scroll_notification_observer.dart#L224-L227

It does not factor in whether any listener it passes the notification to wants to return true and stop the notification from continuing up the tree.

Do you think the ScrollNotificationObserver should include the return value from its listeners?

Reply:
That's a great catch! I'm unsure if it could be an optional return or always return value from listeners.